### PR TITLE
[SDTEST-592] Add documentation for GitLab+CircleCI autoinstrumentation

### PR DESCRIPTION
**Pull request by ManuelPalenzuelaDD**
### What does this PR do? What is the motivation?
This patch adds documentation for GitLab and CircleCI test visibility autoinstrumentation. This also unifies this new docs with the existing GitHub actions + Jenkins autoinstrumentation docs.

**This won't be merged until we've released the CircleCI autoinstrumentation**

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: layouts/shortcodes/ci-autoinstrumentation.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+We support auto-instrumentation for the following CI providers:
+| CI Provider | Auto-Instrumentation method |
+|---|---|
+| GitHub Actions | [Datadog Test Visibility Github Action][1] |
+| Jenkins | [UI-based configuration][2] |
+| GitLab | [Datadog Test Visibility GitLab Script][3] |
+| CircleCI | [Datadog Test Visibility CircleCI Orb][4] |
+
+If you are using autoinstrumentation for one of these providers, the rest of the setup steps below can be skipped.
```
Comment (nikita-tkachenko-datadog) [inline layouts/shortcodes/ci-autoinstrumentation.md:9]:
```suggestion
If you are using auto-instrumentation for one of these providers, the rest of the setup steps below can be skipped.
```

File: layouts/shortcodes/ci-autoinstrumentation.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+We support auto-instrumentation for the following CI providers:
+| CI Provider | Auto-Instrumentation method |
+|---|---|
+| GitHub Actions | [Datadog Test Visibility Github Action][1] |
+| Jenkins | [UI-based configuration][2] |
```
Comment (nikita-tkachenko-datadog) [inline layouts/shortcodes/ci-autoinstrumentation.md:5]:
```suggestion
| Jenkins | [UI-based configuration][2] with Datadog Jenkins plugin |
```

**Review by nikita-tkachenko-datadog — APPROVED** [review]

File: layouts/shortcodes/ci-autoinstrumentation.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+| GitLab | [Datadog Test Visibility GitLab Script][3] |
+| CircleCI | [Datadog Test Visibility CircleCI Orb][4] |
+
+If you are using autoinstrumentation for one of these providers, the rest of the setup steps below can be skipped.
+
+[1]: https://github.com/marketplace/actions/configure-datadog-test-visibility
+[2]: /continuous_integration/pipelines/jenkins/#enable-with-the-jenkins-configuration-ui-1
+[3]: https://github.com/DataDog/test-visibility-gitlab-script
+[4]: https://circleci.com/orbs/registry/orb/datadog/test-visibility-circleci-orb
```
Comment (hestonhoffman) [inline layouts/shortcodes/ci-autoinstrumentation.md:14]: 404s currently, but I'm guessing this is cos it hasn't been released yet?

**Review by hestonhoffman — APPROVED** [review]
LGTM with Nikita's suggestions. Let us know in #documentation when you want to merge!

File: layouts/shortcodes/ci-autoinstrumentation.md (hunk 3 above)
Comment (ManuelPalenzuelaDD) [reply layouts/shortcodes/ci-autoinstrumentation.md:14]: Yes, exactly :)

**Pull request merged**